node-runtime/spec/rest/rest.spec.ts

```diff
@@ -0,0 +1,223 @@
+import { Parser } from "@sdkgen/parser";
+import { generateNodeClientSource, generateNodeServerSource } from "@sdkgen/typescript-generator";
+import axios from "axios";
+import { unlinkSync, writeFileSync } from "fs";
+import { Context, SdkgenHttpServer } from "../../src";
+
+const ast = new Parser(`${__dirname}/api.sdkgen`).parse();
+
+writeFileSync(`${__dirname}/api.ts`, generateNodeServerSource(ast, {}).replace("@sdkgen/node-runtime", "../../src"));
+const { api } = require(`${__dirname}/api.ts`);
+unlinkSync(`${__dirname}/api.ts`);
+
+api.fn.add = async (ctx: Context, { first, second }: { first: number; second: string }) => {
+    return `${first}${second}`;
+};
+
+api.fn.maybe = async (ctx: Context, { bin }: { bin: string | null }) => {
+    return bin !== null ? Buffer.from(bin, "hex") : null;
+};
+
+api.fn.hex = async (ctx: Context, { bin }: { bin: Buffer }) => {
+    return bin.toString("hex");
+};
+
+api.fn.obj = async (ctx: Context, { obj }: { obj: { val: number } }) => {
+    if (obj.val === 0) {
+        throw "Value is zero ~ spec error";
+    }
+    return obj;
+};
+
+writeFileSync(
+    `${__dirname}/nodeClient.ts`,
+    generateNodeClientSource(ast, {}).replace("@sdkgen/node-runtime", "../../src"),
+);
+const { ApiClient: NodeApiClient } = require(`${__dirname}/nodeClient.ts`);
+unlinkSync(`${__dirname}/nodeClient.ts`);
+
+const nodeClient = new NodeApiClient("http://localhost:8001");
+
+const server = new SdkgenHttpServer(api, { aaa: true });
+
+describe("Rest API", () => {
+    beforeAll(() => {
+        server.listen(8001);
+    });
+
+    afterAll(async () => {
+        await server.close();
+    });
+
+    test("add with sdkgen", async () => {
+        expect(await nodeClient.add(null, { first: 1, second: "aa" })).toEqual("1aa");
+    });
+
+    const table = [
```

Comment:
Another test suggestion:

The path `/add2&second=1` should *not* be valid. `&second=1` is not part of query string in this case.

Reply:
Done